### ISSUE TITLE
google_cloudbuild_trigger uses deprecated workerPool field

### ISSUE THREAD

**Issue by mikouaj** [hashicorp/terraform-provider-google#18731]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Terraform Version & Provider Version(s)

Terraform v1.9.0
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v5.35.0
+ provider registry.terraform.io/hashicorp/google-beta v5.35.0

### Affected Resource(s)

google_cloudbuild_trigger

### Terraform Configuration

```tf
resource "google_cloudbuild_trigger" "build_trigger" {
  project     = google_project.project.project_id
  name        = "test-trigger"
  description = "Test trigger"
  location    = var.region

  service_account = google_service_account.build.id

  webhook_config {
    secret = google_secret_manager_secret_version.build_webhook.id
  }

  build {
    step {
      id   = "test"
      name = "gcr.io/cloud-builders/gcloud"
      args = ["version"]
    }
    options {
      worker_pool = google_cloudbuild_worker_pool.pool_01.id
      logging     = "CLOUD_LOGGING_ONLY"
    }
  }
}
```

### Debug Output

https://gist.github.com/mikouaj/4fa1986c5392a749c6d083d3c7e394f8

### Expected Behavior

The provider should use `pool` field with a [PoolOption](https://cloud.google.com/build/docs/api/reference/rest/v1/projects.builds#Build.PoolOption) object value in a [BuildOptions](https://cloud.google.com/build/docs/api/reference/rest/v1/projects.builds#Build.BuildOptions) in a `POST /v1/projects/PROJECT_ID/locations/REGION/triggers` body.

Example:
```
---[ REQUEST ]---------------------------------------
POST /v1/projects/PROJECT_ID/locations/REGION/triggers?alt=json HTTP/1.1
Host: cloudbuild.googleapis.com
User-Agent: Terraform/1.9.0 (+https://www.terraform.io)

Content-Length: 517
Content-Type: application/json
Accept-Encoding: gzip

{
  "build": {
    "options": {
      "dynamicSubstitutions": false,
      "logging": "CLOUD_LOGGING_ONLY",
      "pool": {
          name: "projects/PROJECT_ID/locations/REGION/workerPools/pool-01"
       }
    },
    "steps": [
      {
        "args": [
          "version"
        ],
        "id": "test",
        "name": "gcr.io/cloud-builders/gcloud"
      }
    ],
    "timeout": "600s"
  },
  "description": "Test trigger",
  "name": "test-trigger",
  "serviceAccount": "projects/PROJECT_ID/serviceAccounts/cicd-build@PROJECT_ID.iam.gserviceaccount.com",
  "webhookConfig": {
    "secret": "projects/PROJECT_NUMBER/secrets/build-webhook/versions/1"
  }
}
```

### Actual Behavior

The provided uses deprecated `workerPool` field in a [BuildOptions](https://cloud.google.com/build/docs/api/reference/rest/v1/projects.builds#Build.BuildOptions) in a `POST /v1/projects/PROJECT_ID/locations/REGION/triggers` body.

Example:
```
---[ REQUEST ]---------------------------------------
POST /v1/projects/PROJECT_ID/locations/REGION/triggers?alt=json HTTP/1.1
Host: cloudbuild.googleapis.com
User-Agent: Terraform/1.9.0 (+https://www.terraform.io)

Content-Length: 517
Content-Type: application/json
Accept-Encoding: gzip

{
  "build": {
    "options": {
      "dynamicSubstitutions": false,
      "logging": "CLOUD_LOGGING_ONLY",
      "workerPool": "projects/PROJECT_ID/locations/REGION/workerPools/pool-01"
    },
    "steps": [
      {
        "args": [
          "version"
        ],
        "id": "test",
        "name": "gcr.io/cloud-builders/gcloud"
      }
    ],
    "timeout": "600s"
  },
  "description": "Test trigger",
  "name": "test-trigger",
  "serviceAccount": "projects/PROJECT_ID/serviceAccounts/cicd-build@PROJECT_ID.iam.gserviceaccount.com",
  "webhookConfig": {
    "secret": "projects/PROJECT_NUMBER/secrets/build-webhook/versions/1"
  }
}
```

### Steps to reproduce

1. `terraform apply`

**Comment by ggtisc** [comment]
Hi @mikouaj !

After testing the functionality of the `google_cloudbuild_trigger` everything works as expected without errors. This looks more like a proposal of enhancement or a change in the API than a bug. Are you looking for a change for the argument `options` to use a new `pool` argument instead of the current `worker_pool` argument?

**Comment by mikouaj** [comment]
Hi @ggtisc,
That is correct, I'm looking for a change for the argument `options` to use new `pool` argument istead of the current `worker_pool` argument. But there is a good reason for this, the current `worker_pool` argument is not only deprecated in the API but the trigger created with this argument is not taking it into consideration at all. 

TL;DR; is that use of current, deprecated `worker_pool` argument leads to misconfigured Cloud Build trigger resource that does not work as expected and executes builds in a default pool.

**Comment by ggtisc** [comment]
It is confirmed that this is not a bug but rather it is an enhancement proposal to add a new argument(`pool`) since the current field `worker_pool` cannot be changed since it fulfills other functions.

**Comment by mikouaj** [comment]
> It is confirmed that this is not a bug but rather it is an enhancement proposal 

@ggtisc I would argue about that because the Cloud Build trigger with `worker_pool` created by the provider is currently not working as expected. This has impact on Google Cloud customers but lets handle that internally.
